### ISSUE TITLE
Media Uploads: Uploading .zip Files in Calypso Fails With Error "file could not be uploaded because an error occurred while uploading"

### ISSUE THREAD

**Issue by upwardmomentum84** [Automattic/wp-calypso#80816]
### Quick summary

On an atomic site, when uploading a ZIP file via the Calypso Media Library, it fails with this error "file could not be uploaded because an error occurred while uploading."

The issue does not happen on simple sites such as sites with the Premium plan but only on Atomic sites.

Reported here: 6687917-zd-a8c

Screen recording of the issue:

https://github.com/Automattic/wp-calypso/assets/93800099/2cc0b441-9a79-4d78-af47-766a506768ac

### Steps to reproduce

1. On an Atomic site, go to Media Library.
2. Make sure you are in Default(Calypso) view and not Classic(wp-admin) view.
3. Upload a zip file.
4. Observe the error.

### What you expected to happen

The file to upload successfully without error.

### What actually happened

File doesn't upload but shows error instead.

### Impact

Some (< 50%)

### Available workarounds?

Yes, easy to implement

### Platform (Simple and/or Atomic)

Atomic

**Comment by cuemarie** [comment]
📌 SCRUBBING : RESULT - *Replicated / Could Not Replicate / Uncertain*
- Tested on Atomic – Could Not Replicate

📌 FINDINGS/SCREENSHOTS/VIDEO
- On my first test, I uploaded a ZIPped image file to my AT site via https://wordpress.com/media/SITE_ID, and it uploaded as expected:

#### Screenshots/Recordings

https://github.com/Automattic/wp-calypso/assets/27249804/dc20d18e-f246-4062-9e54-518b586dd4bd

📌 ACTIONS
- Requested author feedback

📌 Message to Author
- @upwardmomentum84 is the ZIP file you used the file that the Customer was trying to upload, or a test file? Can you inspect the contents and advise what it contains, so we can get more specific with this test?

**Comment by upwardmomentum84** [comment]
> is the ZIP file you used the file that the Customer was trying to upload, or a test file? Can you inspect the contents and advise what it contains, so we can get more specific with this test?

Yes, I was testing using one that they had in their media library. I tested again using other sample zip files (some I created) and they upload as expected. It looks like the user's zip file came from https://better-mouse.com/. I tried downloading a file from this site and uploading and the issue occurs. Sounds like the issue is dependent on the specific zip file.

**Comment by cuemarie** [comment]
>It looks like the user's zip file came from https://better-mouse.com/. I tried downloading a file from this site and uploading and the issue occurs. Sounds like the issue is dependent on the specific zip file.

Agreed! In the report, you mentioned:

> The issue does not happen on simple sites such as sites with the Premium plan but only on Atomic sites.

Did the specific zip file from that source upload successfully to a simple site for you? If so, that might be worth poking at further, but if ZIP is being blocked on simple and AT, it seems that's working as expected.

**Comment by upwardmomentum84** [comment]
> Did the specific zip file from that source upload successfully to a simple site for you? If so, that might be worth poking at further, but if ZIP is being blocked on simple and AT, it seems that's working as expected.

I did some more testing using various zip files:
- The zip file from [Better-Mouse](https://better-mouse.com/)
- Unzipped the one from Better-Mouse and then re-zipped to a new zip file.
- Zipped a few random files from my computer including one application to match the Better-Mouse file.
- Tried a few random sample zip files from other sources.

On my Premium site, all zip files uploaded without issue.

On my atomic Business site, only the Better-Mouse zip and the re-zipped version failed. All other zip files upload without issue.

**Comment by cuemarie** [comment]
>On my Premium site, all zip files uploaded without issue.
>
>On my atomic Business site, only the Better-Mouse zip and the re-zipped version failed. All other zip files upload without issue.

Hm, interesting. @Automattic/reactor , it seems like we're blocking certain zipped files from being uploaded to AT sites but not Simple sites. Is this expected behavior? See @upwardmomentum84 's most recent testing for an example 🤔

**Comment by cuemarie** [comment]
I've asked for insight here: p1693438480234909-slack-C036E4KFP2L

**Comment by JoshuaGoode** [comment]
Looks related to https://github.com/Automattic/wp-calypso/issues/50928

What's in the .zip?

**Comment by upwardmomentum84** [comment]
The Better-Mouse zip file just contains a Mac application package. I also tested using the [Cyberduck](https://cyberduck.io/download/) install file which is packaged very similarly and it uploads without error. The issue seems specific to the Better-Mouse file.

**Comment by cuemarie** [comment]
📌 ACTIONS
- Closing in favor of https://github.com/Automattic/wp-calypso/issues/81371
